assets/src/edit-story/app/rightClickMenu/provider.js

```diff
@@ -276,26 +306,58 @@ function RightClickMenuProvider({ children }) {
       targets: [selectedElement.id],
     }));
 
-    // Add styles and animations to element
-    updateElementsById({
-      elementIds: [selectedElement.id],
-      properties: (currentProperties) =>
-        updateProperties(
-          currentProperties,
-          {
-            ...copiedElement.styles,
-            animation: oldAnimationToDelete,
-          },
-          /* commitValues */ true
-        ),
-    });
     addAnimations({ animations: newAnimations });
+
+    // Text elements need the text styles extracted from content before
+    // applying to the other text
+    if (copiedElement.type === 'text' && copiedElement.styles.content) {
+      const { textStyles } = getTextPresets(
+        [copiedElement.styles],
+        {
+          textStyles: [],
+          colors: [],
+        },
+        PRESET_TYPES.STYLE
+      );
+      const { colors } = getTextPresets(
+        [copiedElement.styles],
+        {
+          textStyles: [],
+          colors: [],
+        },
+        PRESET_TYPES.Color
+      );
+      const { content, ...copiedElementStyles } = copiedElement.styles;
+      const updatedElementStyles = {
+        ...copiedElementStyles,
+        ...textStyles[0],
+        ...colors[0].color,
+        animation: oldAnimationToDelete,
+        border: copiedElementStyles.border || null,
+      };
+      handleApplyStyle(updatedElementStyles);
+    } else {
+      // Add styles and animations to element
+      updateElementsById({
+        elementIds: [selectedElement.id],
+        properties: (currentProperties) =>
+          updateProperties(
+            currentProperties,
+            {
+              ...copiedElement.styles,
+              animation: oldAnimationToDelete,
+            },
+            /* commitValues */ true
+          ),
+      });
+    }
```

Comment:
This works great man
<img src="https://user-images.githubusercontent.com/22185279/128579271-42797460-d86f-4958-93e7-54a429ad998c.gif" height="350px" />